File: C++/프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr, vector<vector<int>> queries) {
    vector<int> answer;
    
    int max;
    for(vector<int> query : queries) {
        max = 1000001;
        for(int i = query[0]; i <= query[1]; i++) {
            if(query[2] < arr[i] && arr[i] < max) {
                max = arr[i];
            }
        }
        answer.push_back((max == 1000001 ? -1 : max));
    }
    return answer;
}

File: Python3/프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    answer = [];
    for q in queries:
        min_num = 1000001
        
        for i in range(q[0], q[1] + 1):
            if q[2] < arr[i] and arr[i] < min_num: min_num = arr[i]
            
        answer += [-1 if min_num == 1000001 else min_num]
        
    return answer